homeassistant/components/sensor/bmw_connected_drive.py

```diff
@@ -0,0 +1,98 @@
+"""
+Reads vehicle status from BMW connected drive portal.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.bmw_connected_drive/
+"""
+import logging
+import asyncio
+
+from homeassistant.components.bmw_connected_drive import DOMAIN as BMW_DOMAIN
+from homeassistant.helpers.entity import Entity
+
+DEPENDENCIES = ['bmw_connected_drive']
+
+_LOGGER = logging.getLogger(__name__)
+
+LENGTH_ATTRIBUTES = [
+    'remaining_range_fuel',
+    'mileage',
+    ]
+
+VALID_ATTRIBUTES = LENGTH_ATTRIBUTES + [
+    'remaining_fuel',
+]
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the BMW sensors."""
+    accounts = hass.data[BMW_DOMAIN]
+    _LOGGER.debug('Found BMW accounts: %s',
+                  ', '.join([a.name for a in accounts]))
+    devices = []
+    for account in accounts:
+        for vehicle in account.account.vehicles:
+            for sensor in VALID_ATTRIBUTES:
+                device = BMWConnectedDriveSensor(account, vehicle, sensor)
+                devices.append(device)
+    return add_devices(devices)
```

Comment:
Remove the return. Just call `add_devices(devices, True)` to have the `update` method be called during entity addition.

Reply:
`add_devices(devices, True)` resulted in some wierd "no entity_id set" errors.
But `add_devices(devices)' (without true) works just fine.

Reply:
Ok, the entity_id will not be set during the update before add operation. All other attributes will be available though. Either remove the logging about entity_id in `update`, or do the schedule the update from `async_added_to_hass`.